### PR TITLE
[239882] Prepare for upgrading Elixir to 1.12

### PR DESCRIPTION
**Pull request by Double-oxygeN**
https://acsmine.tok.access-company.com/redmine/issues/239882

### CODE REVIEW

**Review by aYukiSekiguchi — CHANGES_REQUESTED** [review]

File: lib/util/crypto.ex [hunk 1]
```diff
@@ -15,8 +15,8 @@ defmodule Antikythera.Crypto do
   end
 
   defp secure_compare_impl(<<x, left::binary>>, <<y, right::binary>>, acc) do
-    use Bitwise
-    secure_compare_impl(left, right, acc ||| x ^^^ y)
+    use Bitwise, skip_operators: true
+    secure_compare_impl(left, right, bxor(bor(acc, x), y))
```
[the remark this answers is not in the document]
Comment (aYukiSekiguchi) [reply lib/util/crypto.ex:19]: I found the document
https://hexdocs.pm/elixir/1.11.4/operators.html#operator-precedence-and-associativity

`^^^` have higher precedence than `|||`, so we should call `bxor/2` first then `bor/2`.

[the remark this answers is not in the document]
Comment (aYukiSekiguchi) [reply lib/util/crypto.ex:19]: We should add test cases which use `acc` in https://github.com/access-company/antikythera/blob/master/test/lib/util/crypto_test.exs#L6 like:
```elixi
    assert Crypto.secure_compare("abc", "abc")
    refute Crypto.secure_compare("abc", "aac")
```

Your implementation fails at the second test case.

[the remark this answers is not in the document]
Comment (Double-oxygeN) [reply lib/util/crypto.ex]: Sorry, I had a mistake about the operator precedence.
Fixed at aa6ca5b.

File: core/web/template_engine.ex [hunk 2]
```diff
@@ -40,10 +40,18 @@ defmodule AntikytheraCore.TemplateEngine do
     {:__block__, [], Enum.reverse([{:safe, q} | dynamic])}
   end
 
-  @impl true
-  def handle_text(state, text) do
-    %{iodata: iodata} = state
-    %{state | iodata: [text | iodata]}
+  if Version.match?(System.version(), "~> 1.12") do
```
[the remark this answers is not in the document]
Comment (Double-oxygeN) [reply core/web/template_engine.ex:44]: Added a comment at 15c565c.

**Review by aYukiSekiguchi — APPROVED** [review]
LGTM

**Pull request merged**